File: app/src/main/java/jzy/easybind/Japp.java
package jzy.easybind;

import android.app.Activity;
import android.os.Bundle;

import com.blueprint.LibApplication;

/**
 * @another 江祖赟
 * @date 2018/4/24.
 */
public class Japp extends LibApplication {

    @Override
    public void onCreate() {
        super.onCreate();
    }

    @Override
    protected void onApplicationCreate() {

    }

    @Override
    protected boolean configDebugState() {
        return true;
    }

    @Override
    protected String[] configCreshWrapperHighlight() {
        return new String[0];
    }

    @Override
    protected void registerOnActivityCreated(Activity activity, Bundle savedInstanceState) {

    }

    @Override
    protected void registerOnActivityDestroyed(Activity activity) {

    }

    @Override
    protected void registerOnTrimMemory(int level) {

    }

    @Override
    protected String configBaseUrl() {
        return null;
    }
}

File: juimodel/src/main/java/com/yun/juimodel/JApp.kt
package com.yun.juimodel

import android.app.Activity
import android.content.res.Configuration
import android.os.Bundle
import com.blueprint.LibApplication
import com.blueprint.helper.StatusBarHelper
import com.facebook.drawee.backends.pipeline.Fresco

/**
 * @another 江祖赟
 * @date 2018/1/16.
 */
class JApp : LibApplication() {
    override fun configBaseUrl(): String {
        return "666"
    }

    override fun onApplicationCreate() {
        Fresco.initialize(this)
    }

    override fun configCreshWrapperHighlight(): Array<String> {
        return arrayOf()
    }

    override fun registerOnActivityDestroyed(activity: Activity?) {
        println("===========")
    }

    override fun registerOnActivityStopped(activity: Activity?) {
        println("===========")
    }

    override fun registerOnActivityResumed(activity: Activity?) {
    }

    override fun registerOnActivityCreated(activity: Activity?, savedInstanceState: Bundle?) {
        StatusBarHelper.StatusBarLightMode(activity)
    }

    override fun registerOnLowMemory() {
    }

    override fun registerOnConfigurationChanged(newConfig: Configuration?) {
    }

    override fun registerOnTrimMemory(level: Int) {
    }

    override fun onhandlerException(thread: Thread?, throwable: Throwable?): Boolean {
        println("===========")
        return false
    }

    override fun configDebugState(): Boolean {
        return BuildConfig.DEBUG
    }
}